app.py

```diff
@@ -106,29 +112,35 @@ def save_arrangement():
         else:
             mdb.add_arrangement(data)
```

Comment:
I think you need to find the old arrangement if the arrangement already exists in mdb